contrib/core-contract-tests/tests/pox-4/pox_CheckBalanceCommand.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  logCommand,
+  PoxCommand,
+  Real,
+  Stub,
+  Wallet,
+} from "./pox_CommandModel.ts";
+import { expect } from "vitest";
+import { ClarityValue, cvToValue } from "@stacks/transactions";
+
+/**
+ * Implements the `PoxCommand` interface to check a wallet's balance.
+ */
+export class CheckBalanceCommand implements PoxCommand {
```

Comment:
This command looks great, and now that I'm looking into it, we shouldn't need it; instead all we need is to add this in the existing `GetStxAccountCommand`:

```ts
expect(actual.amountLocked + actual.amountUnlocked).toBe(
  actual.ustxBalance,
);
```

Right there:

```diff
diff --git a/contrib/core-contract-tests/tests/pox-4/pox_GetStxAccountCommand.ts \
           b/contrib/core-contract-tests/tests/pox-4/pox_GetStxAccountCommand.ts
index ac5a482b8..73371d6a0 100644
--- a/contrib/core-contract-tests/tests/pox-4/pox_GetStxAccountCommand.ts
+++ b/contrib/core-contract-tests/tests/pox-4/pox_GetStxAccountCommand.ts
@@ -39,6 +39,10 @@ export class GetStxAccountCommand implements PoxCommand {
         "unlock-height": Cl.uint(actual.unlockHeight),
       });

+    expect(actual.amountLocked + actual.amountUnlocked).toBe(
+      actual.ustxBalance,
+    );
+
     // Log to console for debugging purposes. This is not necessary for the
     // test to pass but it is useful for debugging and eyeballing the test.
     logCommand(
```